### PR TITLE
Chore: Update Cuetsy to v0.1.8 version

### PR DESCRIPTION
**Pull request by spinillos**
It updates `Cuetsy` with the latest v0.1.8 version. I know that the branch says v0.1.7 but we had a small bug in that version and we needed to create a new version 😞. 

It also updates `Cuelang` dependency in go.mod and bingo to the latest v0.5.0 stable version.

It includes the following improvement and fixes:

**Improvements**:
* It allows the use of recursive values.
* It allows `null` values.
* It allows enum unions.
* It allows nested default values.

**Fixes**:
* Types and Enums from parent value can be override.
* Inline comments are now working.
* Defaults for Enums and Types are fixed following the Cuelang standard. 
  * We were incorrectly allowing the pattern `#Enum | *"a"` to set a default enum value. It should be a union of elements.
  * The correct way is `#Enum & (*"a" | _)`. It's less intuitive but its the way to differentiate it from the unions. The PR updates these use cases.
  * Its the same pattern for Types.

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/dataquery.gen.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export interface QueryEditorArrayExpression {
   /**
    * TODO this doesn't work; temporarily extended in veneer
    */
-  type: QueryEditorExpressionType;
+  type: (QueryEditorExpressionType.And | QueryEditorExpressionType.Or);
```
Comment (Clarity-89) [inline public/app/plugins/datasource/cloudwatch/dataquery.gen.ts:149]: We'd probably remove the TODO and remove the manual type from veneer.

File: public/app/plugins/panel/heatmap/panelcfg.gen.ts [hunk 2]
```diff
@@ -51,15 +51,15 @@ export interface HeatmapColorOptions {
   /**
    * Sets the color mode
    */
-  mode?: HeatmapColorMode;
+  mode?: HeatmapColorMode; // TODO: remove optional when https://github.com/grafana/cuetsy/issues/74 is fixed
```
Comment (Clarity-89) [inline public/app/plugins/panel/heatmap/panelcfg.gen.ts:54]: Not sure where these comments are coming from, but it looks like the linked issue has been fixed, so the comment is not necessary anymore.

**Review by Clarity-89 — COMMENTED** [review]
Noticed there are a few TODO comments appeared, which have already bee fixed.

File: public/app/plugins/panel/heatmap/panelcfg.gen.ts (hunk 2 above)
Comment (sdboyer) [reply public/app/plugins/panel/heatmap/panelcfg.gen.ts:54]: the PR also fixed inline comment generation, so it's a comment that was already in the .cue file that's now showing up here

**Review by Clarity-89 — APPROVED** [review]
Looks good! I can make a follow-up PR to remove the manual types for the bugs that were fixed.

**Pull request merged**